In [1]:
# Utils
# =============================================================================
from utils.paths import (
    ORDERS_FILE, 
    ORDER_ITEMS_FILE, 
    PAYMENTS_FILE, 
    REVIEWS_FILE, 
    CUSTOMERS_FILE, 
    PRODUCTS_FILE, 
    SELLERS_FILE, 
    GEOLOCATION_FILE, 
    CATEGORY_TRANSLATION_FILE
)

# DATA
# =============================================================================
import pandas as pd

# EDA
# =============================================================================
from data_assessmen.data_inspection import (
    dataset_summary,
    dataset_columns)

# **Data**
---

**`OLIST_FILES`**

A dictionary that maps each table's short internal name (used throughout the code) to its actual CSV filename as distributed by Kaggle.

Keeping this mapping in one dictionary avoids hardcoding filenames in multiple places, and lets the loader iterate over all tables with a single loop instead of repeating near-identical code nine times.

| Key | File | Description |
|---|---|---|
| `orders` | `olist_orders_dataset.csv` | Core orders table; source of order status and timestamps |
| `order_items` | `olist_order_items_dataset.csv` | Line items linked to each order, at product level |
| `payments` | `olist_order_payments_dataset.csv` | Payment records linked to each order |
| `reviews` | `olist_order_reviews_dataset.csv` | Customer reviews linked to each order |
| `customers` | `olist_customers_dataset.csv` | Customer records linked to each order |
| `products` | `olist_products_dataset.csv` | Product catalog linked to order items |
| `sellers` | `olist_sellers_dataset.csv` | Seller records linked to order items |
| `geolocation` | `olist_geolocation_dataset.csv` | Reference table of zip-code coordinates |
| `category_translation` | `product_category_name_translation.csv` | Reference table translating product categories |

**orders**

In [2]:
data = pd.read_csv(ORDERS_FILE)

In [29]:
dataset_summary(data)

DATASET OVERVIEW
────────────────────────────────────────

Rows                         71
Columns                      2

VARIABLE TYPES
────────────────────────────────────────

Categorical           2   ██████████████████████████████████████████████████  100.0%

────────────────────────────────────────


In [4]:
dataset_columns(data)

DATASET COLUMNS
───────────────────────────────────────────────────────
Column              Type              Records     Nulls
───────────────────────────────────────────────────────
order_id            Categorical        99,441         0
customer_id         Categorical        99,441         0
order_status        Categorical        99,441         0
order_purchase_ti...Categorical        99,441         0
order_approved_at   Categorical        99,281       160
order_delivered_c...Categorical        97,658     1,783
order_delivered_c...Categorical        96,476     2,965
order_estimated_d...Categorical        99,441         0
───────────────────────────────────────────────────────


- The dataset contains 99,441 records and 8 variables, with no missing values across the core fields. Missing data is mainly concentrated in order approval and delivery timestamps, particularly in order_delivered_customer_date.

- The timestamp fields are currently classified as categorical variables, so they should be converted to datetime format to support accurate time-based and delivery analysis.

- Overall, the dataset provides high data completeness and a suitable structure for analyzing the order purchase and fulfillment process.

**order items**

In [5]:
data = pd.read_csv(ORDER_ITEMS_FILE)

In [6]:
dataset_summary(data)

DATASET OVERVIEW
────────────────────────────────────────

Rows                         112,650
Columns                      7

VARIABLE TYPES
────────────────────────────────────────

Categorical           4   █████████████████████████████   57.1%
Numeric               2   ██████████████         28.6%
Integer               1   ███████                14.3%

────────────────────────────────────────


In [7]:
dataset_columns(data)

DATASET COLUMNS
───────────────────────────────────────────────────────
Column              Type              Records     Nulls
───────────────────────────────────────────────────────
order_id            Categorical       112,650         0
order_item_id       Integer           112,650         0
product_id          Categorical       112,650         0
seller_id           Categorical       112,650         0
shipping_limit_date Categorical       112,650         0
price               Numeric           112,650         0
freight_value       Numeric           112,650         0
───────────────────────────────────────────────────────


- The dataset contains 112,650 records and 7 variables, with no missing values across any of the fields. The structure combines categorical, numeric, and integer variables covering order items, products, sellers, and transaction-level values.

- The shipping_limit_date field is currently classified as categorical and should be converted to datetime format to enable accurate shipping and time-based analysis.

**payments**

In [8]:
data = pd.read_csv(PAYMENTS_FILE)

In [9]:
dataset_summary(data)

DATASET OVERVIEW
────────────────────────────────────────

Rows                         103,886
Columns                      5

VARIABLE TYPES
────────────────────────────────────────

Categorical           2   ████████████████████   40.0%
Integer               2   ████████████████████   40.0%
Numeric               1   ██████████             20.0%

────────────────────────────────────────


In [10]:
dataset_columns(data)

DATASET COLUMNS
───────────────────────────────────────────────────────
Column              Type              Records     Nulls
───────────────────────────────────────────────────────
order_id            Categorical       103,886         0
payment_sequential  Integer           103,886         0
payment_type        Categorical       103,886         0
payment_installmentsInteger           103,886         0
payment_value       Numeric           103,886         0
───────────────────────────────────────────────────────


- The dataset contains 103,886 records and 5 variables, with no missing values across the dataset. The structure captures payment methods, payment sequence, installment count, and transaction value.

**reviews**

In [30]:
data = pd.read_csv(REVIEWS_FILE)

In [31]:
dataset_summary(data)

DATASET OVERVIEW
────────────────────────────────────────

Rows                         99,224
Columns                      7

VARIABLE TYPES
────────────────────────────────────────

Categorical           6   ███████████████████████████████████████████   85.7%
Integer               1   ███████                14.3%

────────────────────────────────────────


In [13]:
dataset_columns(data)

DATASET COLUMNS
───────────────────────────────────────────────────────
Column              Type              Records     Nulls
───────────────────────────────────────────────────────
review_id           Categorical        99,224         0
order_id            Categorical        99,224         0
review_score        Integer            99,224         0
review_comment_titleCategorical        11,568    87,656
review_comment_me...Categorical        40,977    58,247
review_creation_dateCategorical        99,224         0
review_answer_tim...Categorical        99,224         0
───────────────────────────────────────────────────────


- The dataset contains 99,224 records and 7 variables, with complete coverage for review_score and the review date fields. The data supports the analysis of order ratings and customer feedback through scores and written reviews.

- A high level of missingness is observed in the text-based fields: review_comment_title contains 87,656 missing values, while review_comment_message contains 58,247. This indicates that written feedback is only available for a subset of the reviews.

**customers**

In [14]:
data = pd.read_csv(CUSTOMERS_FILE)

In [15]:
dataset_summary(data)

DATASET OVERVIEW
────────────────────────────────────────

Rows                         99,441
Columns                      5

VARIABLE TYPES
────────────────────────────────────────

Categorical           4   ████████████████████████████████████████   80.0%
Integer               1   ██████████             20.0%

────────────────────────────────────────


In [16]:
dataset_columns(data)

DATASET COLUMNS
───────────────────────────────────────────────────────
Column              Type              Records     Nulls
───────────────────────────────────────────────────────
customer_id         Categorical        99,441         0
customer_unique_id  Categorical        99,441         0
customer_zip_code...Integer            99,441         0
customer_city       Categorical        99,441         0
customer_state      Categorical        99,441         0
───────────────────────────────────────────────────────


- The dataset contains 99,441 records and 5 variables, with no missing values across the dataset. The available fields provide customer-level identification and geographic attributes, including city, state, and zip code.

- Overall, the dataset shows complete and consistent coverage, providing a solid foundation for customer segmentation and geographic analysis.

**products**

In [17]:
data = pd.read_csv(PRODUCTS_FILE)

In [18]:
dataset_summary(data)

DATASET OVERVIEW
────────────────────────────────────────

Rows                         32,951
Columns                      9

VARIABLE TYPES
────────────────────────────────────────

Numeric               7   ███████████████████████████████████████   77.8%
Categorical           2   ███████████            22.2%

────────────────────────────────────────


In [19]:
dataset_columns(data)

DATASET COLUMNS
───────────────────────────────────────────────────────
Column              Type              Records     Nulls
───────────────────────────────────────────────────────
product_id          Categorical        32,951         0
product_category_...Categorical        32,341       610
product_name_lenght Numeric            32,341       610
product_descripti...Numeric            32,341       610
product_photos_qty  Numeric            32,341       610
product_weight_g    Numeric            32,949         2
product_length_cm   Numeric            32,949         2
product_height_cm   Numeric            32,949         2
product_width_cm    Numeric            32,949         2
───────────────────────────────────────────────────────


- The dataset contains 32,951 records and 9 variables, with a strong presence of numeric fields. The data captures key product attributes, including category, dimensions, weight, photo count, and descriptive content.

- A total of 610 missing values are observed across category and description-related fields, while weight and dimension fields contain only 2 missing records. Overall, the dataset provides solid coverage for product attribute analysis.

**sellers**

In [20]:
data = pd.read_csv(SELLERS_FILE)

In [21]:
dataset_summary(data)

DATASET OVERVIEW
────────────────────────────────────────

Rows                         3,095
Columns                      4

VARIABLE TYPES
────────────────────────────────────────

Categorical           3   ██████████████████████████████████████   75.0%
Integer               1   ████████████           25.0%

────────────────────────────────────────


In [22]:
dataset_columns(data)

DATASET COLUMNS
───────────────────────────────────────────────────────
Column              Type              Records     Nulls
───────────────────────────────────────────────────────
seller_id           Categorical         3,095         0
seller_zip_code_p...Integer             3,095         0
seller_city         Categorical         3,095         0
seller_state        Categorical         3,095         0
───────────────────────────────────────────────────────


- The dataset contains 3,095 records and 4 variables, with no missing values. The available fields capture seller identifiers and geographic attributes, including city, state, and zip code.

- Overall, the dataset provides complete and consistent coverage, supporting geographic analysis of the seller base.

**geolocation**

In [23]:
data = pd.read_csv(GEOLOCATION_FILE)


In [24]:
dataset_summary(data)

DATASET OVERVIEW
────────────────────────────────────────

Rows                         1,000,163
Columns                      5

VARIABLE TYPES
────────────────────────────────────────

Numeric               2   ████████████████████   40.0%
Categorical           2   ████████████████████   40.0%
Integer               1   ██████████             20.0%

────────────────────────────────────────


In [25]:
dataset_columns(data)

DATASET COLUMNS
───────────────────────────────────────────────────────
Column              Type              Records     Nulls
───────────────────────────────────────────────────────
geolocation_zip_c...Integer         1,000,163         0
geolocation_lat     Numeric         1,000,163         0
geolocation_lng     Numeric         1,000,163         0
geolocation_city    Categorical     1,000,163         0
geolocation_state   Categorical     1,000,163         0
───────────────────────────────────────────────────────


- The dataset contains 1,000,163 records and 5 variables, with no missing values. The data provides geographic attributes including zip code, latitude, longitude, city, and state, supporting location-based and regional analysis.

- The complete coverage across all fields provides a consistent foundation for geographic segmentation and spatial analysis.

**category translation**

In [26]:
data = pd.read_csv(CATEGORY_TRANSLATION_FILE)

In [27]:
dataset_summary(data)

DATASET OVERVIEW
────────────────────────────────────────

Rows                         71
Columns                      2

VARIABLE TYPES
────────────────────────────────────────

Categorical           2   ██████████████████████████████████████████████████  100.0%

────────────────────────────────────────


In [28]:
dataset_columns(data)

DATASET COLUMNS
───────────────────────────────────────────────────────
Column              Type              Records     Nulls
───────────────────────────────────────────────────────
product_category_...Categorical            71         0
product_category_...Categorical            71         0
───────────────────────────────────────────────────────


In [ ]:
dataset_summary(data)

DATASET OVERVIEW
────────────────────────────────────────

Rows                         99,441
Columns                      5

VARIABLE TYPES
────────────────────────────────────────

Categorical           4   ████████████████████████████████████████   80.0%
Integer               1   ██████████             20.0%

────────────────────────────────────────


- The dataset contains 71 records and 2 variables, with no missing values. The data maps product categories between their original labels and translated equivalents, supporting category standardization and consistent analysis across the dataset.